File: 123/SphBessel/rcbesselcb4.m
% Define the range of real values for z
z_values = linspace(0, 14, 100);

% Initialize arrays for drcj values for different n values
n_values = [1, 2, 3];
drcj_values = zeros(length(n_values), length(z_values));

% Calculate drcj for each n and each real value of z
for i = 1:length(n_values)
    n = n_values(i);
    for j = 1:length(z_values)
        z = complex(z_values(j), 0);  % Imaginary part is 0 for this case
        [~, ~, drcj, ~] = rcbesselc(z, n);
        drcj_values(i, j) = drcj(n + 1);  % MATLAB uses 1-based indexing
    end
end

% Plot drcj for different n values as a function of the real part of z
figure;
for i = 1:length(n_values)
    plot(z_values, real(drcj_values(i, :)), 'LineWidth', 1.5, 'DisplayName', ['n = ', num2str(n_values(i))]);
    hold on;
end

xlabel('Real part of z');
ylabel('drcj');
title('Derivative of Riccati-Bessel function (drcj) for z from 0 to 14');
legend('Location', 'best');
grid on;
hold off;

File: SphBessel/rcbesselcb4.py
import numpy as np
import matplotlib.pyplot as plt
from nrcbesselc import rcbesselc  # Assuming rcbesselc is defined in rcbesselc.py

# Define the range of real values for z
z_values = np.linspace(0, 14, 100)

# Initialize arrays for rcj values for different n values
n_values = [1, 2, 3]
rcj_values = np.zeros((len(n_values), len(z_values)), dtype=complex)

# Calculate rcj for each n and each real value of z
for i, n in enumerate(n_values):
    for j, z_real in enumerate(z_values):
        z = complex(z_real, 0)  # Imaginary part is 0 for this case
        rcj, _, _, _ = rcbesselc(z, n)
        rcj_values[i, j] = rcj[n]

# Plot rcj for different n values as a function of the real part of z
plt.figure(figsize=(10, 6))
for i, n in enumerate(n_values):
    plt.plot(z_values, rcj_values[i].real, label=f'n={n}')

plt.xlabel('Real part of z')
plt.ylabel('rcj')
plt.title('rcj for z from 0 to 14, n=1 to 3')
plt.xlim(0, 14)
plt.ylim(-1.5, 1.5)
plt.legend()
plt.grid(True)
plt.show()
